### PR TITLE
Expand, Modify community roles

### PR DESCRIPTION
**Pull request by alisondy**
This PR expands on the roles set out in GOVERNANCE

### Replace the Contributor role, with a member role with new requirements

Provides a stepping stone between starting to contribute, and becoming a flux maintainer

### Attempt to flesh out the requirements on how one becomes a maintainer

Needs feedback from current maintainers on how that works

### Information shifted from governance into community roles

I've tried to shift everything in the roles section into roles doc which states
- what the role is
- how the role is defined
- the requirements
- the responsibilities and privileges

## Feedback welcome and requested
Would like Feedback from current flux Oversight committee and maintainers to see if theres anything that I have missed, or incorrectly assumed / stated

Need more details on what level of github privileges are for maintainer and oversight to add to respective sections

Fixes https://github.com/fluxcd/community/issues/64

### CODE REVIEW

**Review by kingdonb — APPROVED** [review]
LGTM

File: community-roles.md
Comment (alisondy) [reply community-roles.md:37]:
```suggestion
- Sponsored by 2 maintainers. **Note the following requirements for sponsors:**
```

File: community-roles.md
Comment (alisondy) [inline community-roles.md:35]:
```suggestion
- Subscribed to the [flux-dev mailing list](https://lists.cncf.io/g/cncf-flux-dev/join)
```

**Review by hiddeco — COMMENTED** [review]
Promised to respond to this sometime ago, and already shared a lot of thoughts with you in private, but for completeness and transparency will comment them here again.

I think the idea and aim to be more welcoming and give people a "growth ladder" is great!

My main concern with the changes as proposed at present is that it gives a different meaning to the term "contributor", which would create a diversion with the badge that GitHub displays for anyone who has once contributed to the project, and may conflict with the general idea people have around the term. It would also be less-welcoming in a way, as people that now do a drive-by contribution get less recognition.

My proposal would be the following:

* Keep the "contributor" role in the document to acknowledge drive-by contributors, but remove the "defined by" part.
* Add a "sustained contributor" role that gives the elevated rights and has the requirements currently listed under "contributor".

File: community-roles.md
Comment (scottrigby) [reply community-roles.md]: We worked on this today and found the Triage role would be a good fit for this. Also renamed the role to "Project Member" for clarity.

**Comment by scottrigby** [comment]
> My main concern with the changes as proposed at present is that it gives a different meaning to the term "contributor", which would create a diversion with the badge that GitHub displays for anyone who has once contributed to the project, and may conflict with the general idea people have around the term. It would also be less-welcoming in a way, as people that now do a drive-by contribution get less recognition.

@hiddeco we addressed this using different language. WDYT?

**Review by scottrigby — APPROVED** [review]
lgtm. Disclosure: I collaborated on some of this 😉

**Comment by scottrigby** [comment]
Note we need a Supermajority to make material changes to this Governance doc. The only material change to Governance – apart from moving role info to a separate doc – is introducing a "project member" role that has Triage permission. We need a supermajority to approve this. @hiddeco @squaremo @stefanprodan

**Comment by hiddeco** [comment]
This all looks alright to me. :+1:

File: OVERSIGHT.md
Comment (stefanprodan) [inline OVERSIGHT.md:6]:
```suggestion
| Stefan Prodan | `stefanprodan` | `@stefanprodan` | Weaveworks  |
```

**Review by stefanprodan — APPROVED** [review]
LGTM

Thanks @alisondy 🥇

File: OVERSIGHT.md
Comment (hiddeco) [inline OVERSIGHT.md:3]:
```suggestion
| Name | Slack handle | GitHub handle | Affiliation |
```

**Review by squaremo — APPROVED** [review]
I am all for giving people a lower rung on the ladder to step up onto :heart: I'm minorly concerned that there's a lot of ceremony needed, mostly pushed onto the prospective project member. But there has to be _some_ expectation set for the process, and I'm sure we can streamline it a bit later given a bit of experience with it.

**Comment by kingdonb** [comment]
To build upon the goals of this change, I've floated the idea of a new "Bug Scrub" meeting with a similar structure to SIG-CLI's meeting of the same name, in #101

The new member role after this change gets org-wide Triage role, (that's nice, and very helpful for the Bug Scrub concept.)

I wonder if we should have a lesser status than org-wide triage for every repo, but quickly discarded suggesting that idea as it looks like consensus is likely... even though most contributors are not likely to interact with all repos, we shouldn't need to carefully guard the triage role, and heavy management overhead of the FluxCD org is a thing to be avoided.

(So, it's a 👍 for this change from me, "non-binding")

But... if someone was looking to get triage access for just a single repo that they are interested in, I think maybe we should have a defined path for that too, some people will undoubtedly be intimidated by org-wide access (and to @squaremo's point, we could have a slightly lower bar for those members who only want or need the single-repo triage role, or limited scope.)

**Review by hiddeco — APPROVED** [review]
Just discovered I am the one blocking the merge here by not voting. Alison this looks great :medal_sports: 

If you rebase to fix the merge conflicts, and address @squaremo's last nitpicks, I think you can merge :-)

File: community-roles.md
Comment (scottrigby) [reply community-roles.md:63]: Hey, circling back to this. Yes, exactly. We will clarify

Comment (scottrigby) [reply community-roles.md:63]: Fixed

File: community-roles.md
Comment (scottrigby) [reply community-roles.md:129]: Formalized note of future aim to build oversight committee team requirements, and linked to a new issue to track it

**Comment by scottrigby** [comment]
Feedback addressed!

The link checker is failing only because this new document refers to itself, but it will not yet exist in the repo until this PR is merged.
>Errors in community-roles.md
>✗ https://github.com/fluxcd/community/blob/main/community-roles.md (GitHub token not specified. To check GitHub links reliably, use `--github-token` flag / `GITHUB_TOKEN` env var.)

**Comment by scottrigby** [comment]
To get around this minor GH actions check block, I just removed the unnecessary not yet existing link. All Oversight Committee members have approved. We should now be good to merge ✅

**Comment by hiddeco** [comment]
It still needs a rebase before I can press merge :-)

**Pull request merged**
